Sources/StreamChatUI/ChatChannel/ChatChannelVC.swift

```diff
@@ -180,28 +230,19 @@ open class ChatChannelVC: _ViewController,
         )
     }
 
+    open func chatMessageListVCShouldLoadFirstPage(
+        _ vc: ChatMessageListVC
+    ) {
+        channelController.loadFirstPage()
+    }
+
     // MARK: - ChatMessageListVCDelegate
 
     open func chatMessageListVC(
         _ vc: ChatMessageListVC,
         willDisplayMessageAt indexPath: IndexPath
     ) {
-        guard messageListVC.listView.isTrackingOrDecelerating else {
-            return
-        }
-
-        if indexPath.row < messages.count - 10 {
-            return
-        }
-
-        guard !isLoadingPreviousMessages else {
-            return
-        }
-        isLoadingPreviousMessages = true
-
-        channelController.loadPreviousMessages { [weak self] _ in
-            self?.isLoadingPreviousMessages = false
-        }
+        // no-op
```

Comment:
Also here, would be nice to explain why this becomes no-op and where this is handled now.

Reply:
Additionally, this is an `open` function. What if customers have provided their own implementation here?

Reply:
Basically, we were observing the `willDisplayAt` to do pagination. A customer could override this method still for other things, and it will still work. But very unlikely they were overriding our pagination logic, since it is not even possible as far as I'm aware. The View Pagination Logic is now handled by the `ScrollViewPaginationHandlers`. IMO it does not make sense to provide a comment on why this is now a no-op, and what was changed. But we can provide a comment saying that Stream does not implement this method, but as a customer, you can override it if needed. WYDT?

Reply:
I think we should provide more than just no-op, something like use this method if you want to provide additional logic on `willDisplayAt`.